File: app/src/main/java/io/untaek/animal/Hash/CalloutActivity.java
package io.untaek.animal.Hash;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import io.untaek.animal.R;
import io.untaek.animal.component.UserDetailListViewAdapter;
import io.untaek.animal.component.UserDetailListViewItem;
import io.untaek.animal.firebase.PostInTimeline;
import io.untaek.animal.firebase.UserDetail;
import io.untaek.animal.firebase.dummy;

public class CalloutActivity extends AppCompatActivity {
    List<List<PostInTimeline>> userPost = null;
    UserDetail user;
    boolean findFlag = false;
    String userId ;
    int userPosition = 0;

    List<UserDetailListViewItem> itemList;

    public CalloutActivity(String data) {
        userId = data;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_callout_profile);

        userId = getIntent().getExtras().getString("userId");
        userPosition = 0;//이건 무었을 의미?
        for(UserDetail user : dummy.INSTANCE.getUsersDetail()){//더미에서 뭘가져온겨?
            if(user.getId().equals(userId)){
                break;
            }
            userPosition++;
        }

        // itemlist 초기화 함수
        defineItemList(dummy.INSTANCE.getUsersPost().get(userPosition));
        user = dummy.INSTANCE.getUsersDetail().get(userPosition);

        textViewSetText();
        ListView listView = findViewById(R.id.user_detail_listView);

        UserDetailListViewAdapter userDetailListViewAdapter = new UserDetailListViewAdapter(this, itemList);
        listView.setAdapter(userDetailListViewAdapter);
    }

    //
    List<UserDetailListViewItem> defineItemList(List<List<PostInTimeline>> user){
        itemList = new ArrayList<UserDetailListViewItem>();
        UserDetailListViewItem item;

        for(int i = 0; i<user.size(); i++){
            // petName을 이용하여 해당 pet의 게시글 가져오기
            item = new UserDetailListViewItem(user.get(i).get(0).getPetName(),user.get(i));
            itemList.add(item);
        }
        return  itemList;
    }

    private void textViewSetText(){
        TextView userName = findViewById(R.id.user_detail_user_name);
        ImageView userImage = findViewById(R.id.user_detail_user_image);
        TextView postCount = findViewById(R.id.user_detail_post_count);
        TextView followCount = findViewById(R.id.user_detail_follow_count);
        TextView likeCount = findViewById(R.id.user_detail_likes_count);


        userName.setText(user.getUserName());
        userImage.setImageResource(user.getImgResource());
        postCount.setText(""+ user.getPosts());
        followCount.setText(""+ user.getFollows());
        likeCount.setText(""+ user.getTotalLikes());
    }
}

File: app/src/main/java/io/untaek/animal/Hash/CalloutActivity.kt
package io.untaek.animal.Hash

import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.util.Log
import com.bumptech.glide.Glide
import io.untaek.animal.R
import io.untaek.animal.component.UserDetailRecyclerViewAdapter
import io.untaek.animal.firebase.UserDetail
import kotlinx.android.synthetic.main.activity_callout_profile.*
import kotlinx.android.synthetic.main.activity_user_detail.*

class CalloutActivity : AppCompatActivity() {

    private lateinit var user : UserDetail

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_callout_profile)

        Log.e("ㅋㅋㅋ", "이동완료 !")


        intent.getSerializableExtra("data")?.let { d ->
            user = d as UserDetail
            Log.d("users", user.userName)

            var userName = user_detail_user_name
            val userImage = user_detail_user_image
            val postCount  = user_detail_post_count
            val followCount = user_detail_follow_count
            val likeCount = user_detail_likes_count

            Log.e("ㅋㅋㅋ", "user name : "+user.userName)
            Log.e("ㅋㅋㅋ", "user imgUrl : "+user.pictureUrl)
            Log.e("ㅋㅋㅋ", "user totalPosts : "+user.totalPosts)
            Log.e("ㅋㅋㅋ", "user totalLikes : "+user.totalLikes)
            Log.e("ㅋㅋㅋ", "user totalFollows : "+user.totalFollows)

            userName.text = user.userName
            Glide.with(this).load(Uri.parse(user.pictureUrl).toString()).into(userImage)
            postCount.text = user.totalPosts.toString()
            followCount.text= user.totalFollows.toString()
            likeCount.text = user.totalLikes.toString()

            user_detail_listView.layoutManager = GridLayoutManager(this, 4)
            user_detail_listView.adapter = UserDetailRecyclerViewAdapter(this, user)

//            button_go_back_user_detail_detail.setOnClickListener { finish() }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        //viewer.release()
    }
}
